homeassistant/components/bmw_connected_drive/device_tracker.py

```diff
@@ -1,51 +1,101 @@
 """Device tracker for BMW Connected Drive vehicles."""
 import logging
 
-from homeassistant.util import slugify
+from homeassistant.components.device_tracker import SOURCE_TYPE_GPS
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
 
 from . import DOMAIN as BMW_DOMAIN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_scanner(hass, config, see, discovery_info=None):
-    """Set up the BMW tracker."""
-    accounts = hass.data[BMW_DOMAIN]
-    _LOGGER.debug("Found BMW accounts: %s", ", ".join([a.name for a in accounts]))
-    for account in accounts:
-        for vehicle in account.account.vehicles:
-            tracker = BMWDeviceTracker(see, vehicle)
-            account.add_update_listener(tracker.update)
-            tracker.update()
-    return True
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the BMW ConnectedDrive tracker from config entry."""
+    account = hass.data[BMW_DOMAIN][config_entry.entry_id]
+    devices = []
 
+    for vehicle in account.account.vehicles:
+        if vehicle.state.is_vehicle_tracking_enabled:
+            devices.append(BMWDeviceTracker(account, vehicle))
+        else:
+            _LOGGER.info(
+                "Tracking is disabled for vehicle %s (%s)", vehicle.name, vehicle.vin
+            )
+    async_add_entities(devices, True)
 
-class BMWDeviceTracker:
+
+class BMWDeviceTracker(TrackerEntity):
     """BMW Connected Drive device tracker."""
 
-    def __init__(self, see, vehicle):
+    def __init__(self, account, vehicle):
         """Initialize the Tracker."""
-        self._see = see
-        self.vehicle = vehicle
+        self._account = account
+        self._vehicle = vehicle
+        self._unique_id = vehicle.vin
+        self._location = vehicle.state.gps_position
```

Comment:
are we sure this returns a tuple ?

Reply:
Good catch - `vehicle.state.gps_postition` can return `None` if not available. Should we work around this in here or change the upstream library to return `(None, None)`?

Reply:
can't be done here IMHO

Reply:
Done for now in here (see code below). If this is not wanted, I'll update the upstream library.
```
        self._location = (
            vehicle.state.gps_position if vehicle.state.gps_position else (None, None)
        )
```